lib/core/src/chain/bitcoin/service.rs

```diff
@@ -82,147 +76,130 @@ pub trait BitcoinChainService: Send + Sync {
 }
 
 pub(crate) struct HybridBitcoinChainService {
-    client: OnceLock<Client>,
     config: Config,
-    last_known_tip: Mutex<Option<HeaderNotification>>,
+    client: RwLock<Option<BitcoinClient>>,
+    last_known_tip: Mutex<Option<u32>>,
 }
 impl HybridBitcoinChainService {
-    pub fn new(config: Config) -> Result<Self, Error> {
+    pub fn new(config: Config) -> Result<Self> {
         Ok(Self {
             config,
-            client: OnceLock::new(),
+            client: RwLock::new(None),
             last_known_tip: Mutex::new(None),
         })
     }
 
-    fn get_client(&self) -> Result<&Client> {
-        if let Some(c) = self.client.get() {
-            return Ok(c);
+    fn init_client(&self) -> Result<BitcoinClient> {
+        for explorer in &self.config.bitcoin_explorers {
+            if let Ok(client) = BitcoinClient::try_from_explorer(explorer, self.config.network) {
+                if client.is_available() {
+                    return Ok(client);
+                }
+            }
         }
-        let (tls, validate_domain) = match self.config.network {
-            LiquidNetwork::Mainnet | LiquidNetwork::Testnet => (true, true),
-            LiquidNetwork::Regtest => (false, false),
-        };
-        let electrum_url =
-            ElectrumUrl::new(&self.config.bitcoin_electrum_url, tls, validate_domain)?;
-        let client = electrum_url.build_client(&ElectrumOptions { timeout: Some(3) })?;
 
-        let client = self.client.get_or_init(|| client);
-        Ok(client)
+        bail!("Could not create Bitcoin chain service client: no working clients found");
+    }
+
+    async fn set_client(&self) -> Result<()> {
+        let lock = self.client.read().await;
+        if let Some(client) = lock.as_ref() {
+            if client.is_available() {
+                return Ok(());
+            }
+        }
+
+        let mut lock = self.client.write().await;
+        *lock = Some(self.init_client()?);
```

Comment:
This introduces network request inside the lock which will cause it to hang. I suggest to remove the is_available for now.